kaldb/src/main/java/com/slack/kaldb/chunkManager/CachingChunkManager.java

```diff
@@ -95,15 +94,17 @@ protected void shutDown() throws Exception {
   public static CachingChunkManager<LogMessage> fromConfig(
       MeterRegistry meterRegistry,
       MetadataStore metadataStore,
-      KaldbConfigs.ServerConfig serverConfig)
+      KaldbConfigs.ServerConfig serverConfig,
+      KaldbConfigs.S3Config s3Config,
+      KaldbConfigs.CacheConfig cacheConfig)
       throws Exception {
     return new CachingChunkManager<>(
         meterRegistry,
         metadataStore,
-        getS3BlobFsClient(KaldbConfig.get().getS3Config()),
+        getS3BlobFsClient(s3Config),
         SearchContext.fromConfig(serverConfig),
-        KaldbConfig.get().getS3Config().getS3Bucket(),
-        KaldbConfig.get().getCacheConfig().getDataDirectory(),
-        KaldbConfig.get().getCacheConfig().getSlotsPerInstance());
+        s3Config.getS3Bucket(),
```

Comment:
If we only need the `s3Config`, the `dataDirectory`, and `slotsPerInstance` can we just pass those instead?

Reply:
Yeah, I considered it and then decided to pass most general object, so it's more future proof when we add more params to this object.

Reply:
Let's pass the most specific object possible for all of these methods. We do not expect to be changing the behavior of these classes significantly as they are functionally complete at this time. 

It is not a significant amount of work to pass in an additional argument if needed in the future, and it will help when writing and troubleshooting if only the required arguments are passed.

Reply:
This one we agreed to leave for now, and revisit later with potentially a better scoped local config argument.